stix_shifter/stix_transmission/src/modules/stix_bundle/stix_bundle_connector.py

```diff
@@ -42,7 +42,28 @@ def match(self, pattern, observed_data_sdos, verbose=False):
         return matching_sdos
 
     def ping(self):
-        return {"success": True}
+        return_obj = dict()
+
+        bundle_url = self.connection.get('host')
+        auth = self.configuration.get('auth')
+
+        response = self.call_api(bundle_url, auth, 'head')
+        response_code = response.status_code
+
+        if response_code == 200:
+            return_obj['success'] = True
+        elif response_code == 301:
```

Comment:
what I would do here is maybe add default `host` parameter to `ping` and set it to null.
then, when return code is 301, you get the new host value from location and call `ping` again with the new host.